tests/test_streams.py

```diff
@@ -1131,6 +1131,49 @@ async def test_empty_stream_reader_iter_chunks() -> None:
         await iter_chunks.__anext__()
 
 
+async def test_empty_stream_reader_function() -> None:
+    """Test the empty_stream_reader() function creates fresh instances."""
+    reader1 = streams.empty_stream_reader()
+    reader2 = streams.empty_stream_reader()
+
+    # Should be different instances
+    assert reader1 is not reader2
+    assert reader1 is not streams.EMPTY_PAYLOAD
+    assert reader2 is not streams.EMPTY_PAYLOAD
+
+    # Both should start with fresh state
+    assert reader1._read_eof_chunk is False
+    assert reader2._read_eof_chunk is False
+
+
+async def test_empty_stream_reader_no_state_sharing() -> None:
+    """Test that fresh EmptyStreamReader instances don't share state."""
+    reader1 = streams.empty_stream_reader()
+    reader2 = streams.empty_stream_reader()
+
+    # Use reader1 - this will modify its internal state
+    chunks = []
+    async for chunk in reader1.iter_chunks():
+        chunks.append(chunk)
+        if len(chunks) > 5:  # Safety break
+            break
```

Comment:
Coverage doesn't seem happy with these. Not sure if it's just confused, as it seems to suggest that the break line is reached without any of the above lines being reached...

Reply:
This would be resolved by my other suggestion.